modules/repository/src/main/java/org/mule/runtime/module/repository/interal/DefaultRepositoryService.java

```diff
@@ -0,0 +1,134 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.repository.interal;
+
+import org.mule.runtime.module.reboot.MuleContainerBootstrapUtils;
+import org.mule.runtime.module.repository.api.BundleDescriptor;
+import org.mule.runtime.module.repository.api.BundleNotFoundException;
+import org.mule.runtime.module.repository.api.RepositoryConnectionException;
+import org.mule.runtime.module.repository.api.RepositoryService;
+import org.mule.runtime.module.repository.api.RepositoryServiceDisabledException;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+import org.eclipse.aether.transfer.ArtifactNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultRepositoryService implements RepositoryService
+{
+
+    private static final String REPOSITORY_FOLDER = "repository";
+    private static final String DEFAULT_REPOSITORY_TYPE = "default";
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultRepositoryService.class);
+
+    private final RepositorySystem repositorySystem;
+    private final DefaultRepositorySystemSession repositorySystemSession;
+    private List<RemoteRepository> remoteRepositories;
+    private File dependenciesFolder;
+
+    public DefaultRepositoryService()
+    {
+        repositorySystem = new SpiRepositorySystemFactory().createRepositorySystem();
+        createRepositoryFolderIfDoesNotExists();
+        collectRemoteRepositories();
+        repositorySystemSession = new DefaultRepositorySystemSession();
+        repositorySystemSession.setLocalRepositoryManager(repositorySystem.newLocalRepositoryManager(repositorySystemSession, new LocalRepository(dependenciesFolder)));
+    }
+
+    private void collectRemoteRepositories()
+    {
+        String remoteRepositories = System.getProperty(MULE_REMOTE_REPOSITORIES_PROPERTY, "");
+        String[] remoteRepositoriesArray = remoteRepositories.split(",");
+        this.remoteRepositories = new ArrayList<>();
+        for (String remoteRepository : remoteRepositoriesArray)
+        {
+            if (!remoteRepositories.trim().equals(""))
+            {
+                this.remoteRepositories.add(new RemoteRepository.Builder(remoteRepository, DEFAULT_REPOSITORY_TYPE, remoteRepositories.trim()).build());
+            }
+        }
+    }
+
+    private void createRepositoryFolderIfDoesNotExists()
```

Comment:
Seems to me that there are mixed responsibilities in this class, the implementation does not just llokupBundle, but it is also in charge of setup the context (creating folders, searching repositiores, etc). Try to move the creation part of this implementation to a factory